### PR TITLE
nixos/borgmatic: Allow defining multiple configurations

### PR DESCRIPTION
**Pull request by Tom-Hubrecht**
###### Description of changes

Add the option `services.borgmatic.configurations` to allow multiple borgmatic configurations to be defined.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Tom-Hubrecht** [comment]
cc @RaitoBezarius

File: nixos/modules/services/backup/borgmatic.nix [hunk 1]
```diff
@@ -5,54 +5,71 @@ with lib;
...
+          Paths to local or remote repositories (required). Tildes are
+          expanded. Multiple repositories are backed up to in
+          sequence. Borg placeholders can be used. See the output of
+          "borg help placeholders" for details. See ssh_command for
+          SSH options like identity file or port. If systemd service
+          is used, then add local repository paths in the systemd
+          service file to the ReadWritePaths list.
+        '';
+        example = [
+          "user@backupserver:sourcehostname.borg"
```
Comment (RaitoBezarius) [inline nixos/modules/services/backup/borgmatic.nix:32]:
```suggestion
          "user@backupserver:example.org"
```

File: nixos/modules/services/backup/borgmatic.nix [hunk 2]
```diff
@@ -5,54 +5,71 @@ with lib;
...
+        example = [ "/home" "/etc" "/var/log/syslog*" ];
+      };
+      repositories = mkOption {
+        type = listOf str;
+        description = mdDoc ''
+          Paths to local or remote repositories (required). Tildes are
+          expanded. Multiple repositories are backed up to in
+          sequence. Borg placeholders can be used. See the output of
+          "borg help placeholders" for details. See ssh_command for
+          SSH options like identity file or port. If systemd service
```
Comment (RaitoBezarius) [inline nixos/modules/services/backup/borgmatic.nix:27]: Can the systemd service *not* used?

**Review by RaitoBezarius — APPROVED** [review]
The change looks good to me.

File: nixos/modules/services/backup/borgmatic.nix (hunk 1 above)
Comment (imlonghao) [reply nixos/modules/services/backup/borgmatic.nix:32]: That's not an example domain, but a folder name instead.

Maybe we can update this list to an up-to-date version https://github.com/borgmatic-collective/borgmatic/commit/3be99de5b1eb189e0166d5dec8fbe5058fe85bf5

```
ssh://user@backupserver/./sourcehostname.borg
ssh://user@backupserver/./{fqdn}
/var/local/backups/local.borg
```

**Review by imlonghao — APPROVED** [review]
LGTM, local test works.

File: nixos/modules/services/backup/borgmatic.nix (hunk 2 above)
Comment (Tom-Hubrecht) [reply nixos/modules/services/backup/borgmatic.nix:27]: It is possible to disable the `borgmatic` service via `systemd.services.borgmatic.enable = false` and run your own service instead

**Comment by RaitoBezarius** [comment]
Apologies for the delay, @Tom-Hubrecht -- you should have pestered me in the hackerspace :P.

**Pull request merged**